src/Result/task_result.py

```diff
@@ -38,22 +43,36 @@ def pass_rate(self):
         return pass_rate
 
     def execution_duration(self):
-        if not self.has_results() or not self._ended:
+        self._raise_if_not_ended()
+
+        if not self.has_results():
             return 0
 
         return self._ended - self._started
 
+    def _raise_if_ended(self):
+        if self._ended:
+            raise Exception("This task result has already been finalized")
+
+    def _raise_if_not_ended(self):
+        if not self._ended:
+            raise Exception("This task result has not yet been finalized")
+
     def __repr__(self):
         return self.__str__()
 
     def __str__(self):
+        return self.print("")
+
+    def print(self, indent):
         representation = f"{self.name} {TaskResult.TYPE_PREFIX} {self._task_type}"
 
-        if not self.has_results():
-            return representation
+        if not self.enabled:
```

Comment:
A test for this specific case is missing.
Although I agree this is not that important, I would be great to satisfy test coverage :)